test/cartesian/CartesianGrid.spec.tsx

```diff
@@ -30,6 +30,179 @@ describe('<CartesianGrid />', () => {
+      bottom: 1,
+      left: 2,
+      right: 4,
+      top: 8,
+    };
+    it('should put x, y, width and height as coordinates to the background', () => {
+      const { container } = render(
+        <AreaChart margin={exampleMargin} width={500} height={400}>
+          <CartesianGrid fill="green" verticalPoints={verticalPoints} horizontalPoints={horizontalPoints} />
+        </AreaChart>,
+      );
+      const background = container.querySelector('rect.recharts-cartesian-grid-bg');
+      expect.soft(background).toHaveAttribute('x', '2');
+      expect.soft(background).toHaveAttribute('y', '8');
+      expect.soft(background).toHaveAttribute('width', '494');
+      expect.soft(background).toHaveAttribute('height', '391');
+    });
+
+    it('should put x, y, width and height as coordinates to all lines', () => {
+      const { container } = render(
+        <AreaChart margin={exampleMargin} width={500} height={400}>
+          <CartesianGrid fill="green" verticalPoints={verticalPoints} horizontalPoints={horizontalPoints} />
+        </AreaChart>,
+      );
+      const horizontalLines = container.querySelectorAll('.recharts-cartesian-grid-horizontal line');
+      expect(horizontalLines).toHaveLength(5);
+      horizontalLines.forEach(line => {
+        expect.soft(line).toHaveAttribute('x', '2');
+        expect.soft(line).toHaveAttribute('y', '8');
+        expect.soft(line).toHaveAttribute('x1', '2');
+        expect.soft(line).toHaveAttribute('x2', '496');
+      });
+      const verticalLines = container.querySelectorAll('.recharts-cartesian-grid-vertical line');
+      expect(verticalLines).toHaveLength(4);
+      verticalLines.forEach(line => {
+        expect.soft(line).toHaveAttribute('x', '2');
+        expect.soft(line).toHaveAttribute('y', '8');
+        expect.soft(line).toHaveAttribute('y1', '8');
+        expect.soft(line).toHaveAttribute('y2', '399');
+      });
+    });
+
+    it('should put x, y, width and height as coordinates to all stripes', () => {
+      const { container } = render(
+        <AreaChart margin={exampleMargin} width={500} height={400}>
+          <CartesianGrid
+            fill="green"
+            verticalPoints={verticalPoints}
+            horizontalPoints={horizontalPoints}
+            horizontalFill={['red', 'black']}
+            verticalFill={['green', 'blue']}
+          />
+        </AreaChart>,
+      );
+      const horizontalStripes = container.querySelectorAll('.recharts-cartesian-gridstripes-horizontal rect');
+      expect(horizontalStripes).toHaveLength(5);
+      horizontalStripes.forEach(stripe => {
+        expect.soft(stripe).toHaveAttribute('x', '2');
+        expect.soft(stripe).toHaveAttribute('width', '494');
+      });
+      const verticalStripes = container.querySelectorAll('.recharts-cartesian-gridstripes-vertical rect');
+      expect(verticalStripes).toHaveLength(5);
+      verticalStripes.forEach(stripe => {
+        expect.soft(stripe).toHaveAttribute('y', '8');
```

Comment:
Interesting. Would you like to suggest us to use expect.soft always? 
Should we document this as a recommendation? 

I have seen this for the first time, but it does make sense. https://vitest.dev/api/expect.html#soft

Reply:
I find it useful because sometimes I want to see all failures in a test to have a full picture of what is going on - it helps me debug.

I think it's a personal preference so I wouldn't make any recommendations. The tool is there, use it if you like.